task/host-sp-comms/src/main.rs

```diff
@@ -114,45 +255,197 @@ impl ServerImpl {
+                // TODO where do we get host MAC addrs?
+                Some(SpToHost::MacAddresses {
+                    base: [0; 6],
+                    count: 1,
+                    stride: 1,
+                })
+            }
+            HostToSp::HostBootFailure { .. } => {
+                // TODO what do we do in reaction to this? reboot?
+                Some(SpToHost::Ack)
+            }
+            HostToSp::HostPanic { .. } => {
+                // TODO log event and/or forward to MGS
+                Some(SpToHost::Ack)
+            }
+            HostToSp::GetStatus => Some(SpToHost::Status(self.status)),
+            HostToSp::ClearStatus { mask } => {
+                ringbuf_entry!(Trace::ClearStatus { mask });
+                self.status &= Status::from_bits_truncate(!mask);
+                Some(SpToHost::Status(self.status))
+            }
+            HostToSp::GetAlert { mask: _ } => {
+                // TODO define alerts
+                Some(SpToHost::Alert { action: 0 })
+            }
+            HostToSp::RotRequest => {
+                // TODO forward request to RoT; for now just echo
+                response_data = data;
+                Some(SpToHost::RotResponse)
+            }
+            HostToSp::RotAddHostMeasurements => {
+                // TODO forward request to RoT
+                Some(SpToHost::Ack)
+            }
+            HostToSp::GetPhase2Data { start, count: _ } => {
+                // TODO forward real data
+                response_data = b"hello world";
+                Some(SpToHost::Phase2Data { start })
+            }
+        };
+
+        // We set the high bit of the sequence number before responding.
+        header.sequence |= SEQ_REPLY;
+
+        // TODO this can panic if `response_data` is too large; where do we
+        // check it before sending a response?
+        let tx_msg_buf = &mut self.tx_msg_buf; // borrow checker workaround
+        let n = response.map(|response| {
+            host_sp_messages::serialize(
+                tx_msg_buf,
+                &header,
+                &response,
+                response_data,
+            )
+            .unwrap_lite()
+            .0
+        });
+
+        // Now that all buffer borrowing is done, we can borrow `self` mutably
+        // again to perform any necessary action.
+        if let Some(action) = action {
+            match action {
+                Action::RebootHost => self.power_off_host(true),
+                Action::PowerOffHost => self.power_off_host(false),
+            }
+        }
+
+        n
+    }
+}
+
+// approximately idol_runtime::NotificationHandler, in anticipation of
+// eventually having an idol interface (at least for mgmt-gateway to give us
+// host phase 2 data)
+impl ServerImpl {
+    fn handle_notification(&mut self, bits: u32) {
+        if bits & USART_IRQ != 0 {
+            self.handle_usart_notification();
+            sys_irq_control(USART_IRQ, true);
```

Comment:
Do we need to call this during startup to receive the initial interrupt?

Reply:
It was already there, just isn't showing in the diff (line 62).

Reply:
thanks, I was confused how this was working without it!